lib/firestore/models.dart

```diff
@@ -53,20 +56,154 @@ abstract class Reference {
+        operator: StructuredQuery_FieldFilter_Operator.IN);
+    return this;
+  }
+
+  void _createFilter(String fieldPath, dynamic value,
+      {StructuredQuery_FieldFilter_Operator operator}) {
+    var queryFilter = StructuredQuery_Filter();
+    if (value == null || operator == null) {
+      var filter = StructuredQuery_UnaryFilter();
+      filter.op = StructuredQuery_UnaryFilter_Operator.IS_NULL;
+      filter.field_2 = StructuredQuery_FieldReference()..fieldPath = fieldPath;
+
+      queryFilter.unaryFilter = filter;
+    } else {
+      var filter = StructuredQuery_FieldFilter();
+      filter.op = operator;
+      filter.value = _encode(value);
+
+      final fieldReference = StructuredQuery_FieldReference()
+        ..fieldPath = fieldPath;
+      filter.field_1 = fieldReference;
+
+      queryFilter.fieldFilter = filter;
+    }
+    _addToComposite(queryFilter);
+  }
+
+  void _addToComposite(StructuredQuery_Filter filter) {
+    StructuredQuery_CompositeFilter compositeFilter;
+    if (_structuredQuery.hasWhere() &&
+        _structuredQuery.where.hasCompositeFilter()) {
+      compositeFilter = _structuredQuery.where.compositeFilter;
+    } else {
+      compositeFilter = StructuredQuery_CompositeFilter()
+        ..op = StructuredQuery_CompositeFilter_Operator.AND;
+    }
+
+    compositeFilter.filters.add(filter);
+    _structuredQuery.where = StructuredQuery_Filter()
+      ..compositeFilter = compositeFilter;
+  }
+
+  /// Returns [CollectionReference] that's additionally sorted by the specified
+  /// [fieldPath].
+  ///
+  /// The field is a [String] representing a single field name.
+  /// After a [CollectionReference] order by call, you cannot add any more [orderBy]
+  /// calls.
+  CollectionReference orderBy(
+    String fieldPath, {
+    bool descending = false,
+  }) {
+    final order = StructuredQuery_Order();
+    order.field_1 = StructuredQuery_FieldReference()..fieldPath = fieldPath;
+    order.direction = descending
+        ? StructuredQuery_Direction.DESCENDING
+        : StructuredQuery_Direction.ASCENDING;
+    _structuredQuery.orderBy.add(order);
+    return this;
+  }
+
+  /// Returns [CollectionReference] that's additionally limited to only return up
+  /// to the specified number of documents.
+  CollectionReference limit(int count) {
+    _structuredQuery.limit = Int32Value()..value = count;
+    return this;
+  }
+
+  /// Delegates encoding the given [value] to [FirebaseEncoding.encode].
+  fs.Value _encode(dynamic value) {
+    return FirestoreEncoding.encode(value);
+  }
+
   DocumentReference document(String id) {
     return DocumentReference(_gateway, '$path/$id');
   }
 
   Future<Page<Document>> get(
           {int pageSize = 1024, String nextPageToken = ''}) =>
-      _gateway.getCollection(_fullPath, pageSize, nextPageToken);
+      _structuredQuery == null
+          ? _gateway.getCollection(_fullPath, pageSize, nextPageToken)
+          : gateway.runQuery(_structuredQuery, _fullPath);
```

Comment:
`pageSize` and `pageToken` are ignored. Is there no paging when using queries?

Reply:
it's what limit() is for iirc